homeassistant/config_entries.py

```diff
@@ -271,43 +282,43 @@ def __init__(
     ) -> None:
         """Initialize a config entry."""
         # Unique id of the config entry
-        self.entry_id = entry_id or uuid_util.random_uuid_hex()
+        object.__setattr__(self, "entry_id", entry_id or uuid_util.random_uuid_hex())
```

Comment:
We may want to convert to `attrs` in future PR of this as it might make the syntax a bit better

Reply:
I think we should move away completely from using attrs in core. We should use dataclass when appropriate.

Reply:
That makes sense.  The only thing that really gets us here is https://www.attrs.org/en/stable/api.html#attrs.setters.frozen and it probably isn't a good enough justification to use it in a new place when we want to move more to dataclasses

Reply:
The standardlib `dataclasses` is very heavily influenced by `attrs`, I think it's even written by the same author. In my opinion, what we should move away from is using `attrs` when standardlib `dataclasses` is enough.

In this case I think the `attrs.setters.frozen` is enough to motivate using `attrs`.

Reply:
I don't agree. I don't think we should include another library in core that requires us to learn a new API when there is a standard library API that is enough in almost all cases.